pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -373,6 +377,10 @@ func (c *Controller) deleteS3Snapshot(b *v3.EtcdBackup) error {
 	if err != nil {
 		return err
 	}
+	if b.Spec.BackupConfig.S3BackupConfig.EndpointCA != "" {
```

Comment:
shall we check for backup config, and s3 backup config to be not nil too? Or we are safe at this point

Reply:
it's checked few lines earlier https://github.com/rancher/rancher/pull/21020/files/a6ec63d702a8c45b5620e2f9fb5987bc60b1854e#diff-ffdc9a1bfb3f66be2058cc55605965afR349